content/300-guides/300-upgrade-guides/200-upgrading-versions/030-upgrading-to-prisma-4.mdx

```diff
@@ -58,7 +58,17 @@ If you use literal strings to address these values, then you must update your co
 - For `JsonNull` use `Prisma.JsonNull`
 - For `AnyNull`: use `Prisma.AnyNull`
 
-If you already use these named constants, then you do not need to take any action. If you now get a type error when you pass `Prisma.DbNull` as the value of a JSON field, then this probably indicates a bug in your code that our types did not catch before v4.0.0. The field where you tried to store`DbNull is probably not nullable in your schema. As a result, a literal "DbNull" string was stored in the database instead of NULL.
+If you already use these named constants, then you do not need to take any action. If you now get a type error when you pass `Prisma.DbNull` as the value of a JSON field, then this probably indicates a bug in your code that our types did not catch before v4.0.0. The field where you tried to store `DbNull` is probably not nullable in your schema. As a result, a literal "DbNull" string was stored in the database instead of NULL.
+
+### Prisma no longer exports `Prisma.dmmf.schema` into the generated Prisma Client
+
+From v4.0.0, Prisma no longer exports `Prisma.dmmf.schema` into the generated Prisma Client. This makes the generated Prisma Client much more efficient, and also avoids some memory leaks if you use Jest to carry out tests.
+
+Note:
+
+- This change does not affect the DMMF that Prisma passes to the generators.
+- You can retrieve the full DMMF at runtime dynamically.
+- We still export `Prisma.dmmf.datamodel` into the generated Prisma Client.
```

Comment:
This technically was not documented before, so I think we can skip documenting the change as well. (As the state after is also not documented.)

Reply:
Yes. But as this made its way everywhere now (also release notes), I guess I am in the minority on that view.

Reply:
OK - let's leave it in then.